saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -1254,145 +1334,248 @@ scVectorReduced sc ety xs
 scSubNat :: SharedContext -> Term -> Term -> IO Term
 scSubNat sc x y = scGlobalApply sc "Prelude.subNat" [x,y]
 
+-- | Create a term computing the product of the two given (natural number)
+-- terms.
+--
+-- > mulNat : Nat -> Nat -> Nat;
 scMulNat :: SharedContext -> Term -> Term -> IO Term
 scMulNat sc x y = scGlobalApply sc "Prelude.mulNat" [x,y]
 
--- divNat :: Nat -> Nat -> Nat;
+-- | Create a term computing the quotient of the two given (natural number)
+-- terms.
+--
+-- > divNat : Nat -> Nat -> Nat;
 scDivNat :: SharedContext -> Term -> Term -> IO Term
 scDivNat sc x y = scGlobalApply sc "Prelude.divNat" [x,y]
 
--- modNat :: Nat -> Nat -> Nat;
+-- | Create a term computing the remainder upon division of the two given
+-- (natural number) terms.
+--
+-- > modNat : Nat -> Nat -> Nat;
 scModNat :: SharedContext -> Term -> Term -> IO Term
 scModNat sc x y = scGlobalApply sc "Prelude.modNat" [x,y]
 
--- divModNat :: Nat -> Nat -> #(Nat, Nat);
+-- | Create a term computing the quotient and remainder upon division of the
+-- two given (natural number) terms, giving the result as a pair.
+--
+-- > divModNat : Nat -> Nat -> Nat * Nat;
 scDivModNat :: SharedContext -> Term -> Term -> IO Term
 scDivModNat sc x y = scGlobalApply sc "Prelude.divModNat" [x,y]
 
+-- | Create a term computing whether the two given (natural number) terms are
+-- equal.
+--
+-- > equalNat : Nat -> Nat -> Bool;
 scEqualNat :: SharedContext -> Term -> Term -> IO Term
 scEqualNat sc x y = scGlobalApply sc "Prelude.equalNat" [x,y]
 
+-- | Create a term computing whether the first term (a natural number) is less
+-- than the second term (also a natural number).
+--
+-- > ltNat : Nat -> Nat -> Bool;
 scLtNat :: SharedContext -> Term -> Term -> IO Term
 scLtNat sc x y = scGlobalApply sc "Prelude.ltNat" [x,y]
 
+-- | Create a term computing the minimum of the two given (natural number)
+-- terms.
+--
+-- > minNat : Nat -> Nat -> Nat
 scMinNat :: SharedContext -> Term -> Term -> IO Term
 scMinNat sc x y = scGlobalApply sc "Prelude.minNat" [x,y]
 
+-- | Create a term computing the maximum of the two given (natural number)
+-- terms.
+--
+-- > maxNat : Nat -> Nat -> Nat;
 scMaxNat :: SharedContext -> Term -> Term -> IO Term
 scMaxNat sc x y = scGlobalApply sc "Prelude.maxNat" [x,y]
 
 -- Primitive operations on Integer
 
+-- | Create a term representing the prelude Integer type.
 scIntegerType :: SharedContext -> IO Term
 scIntegerType sc = scFlatTermF sc preludeIntegerType
 
+-- | Create an integer constant term from an 'Integer'.
 scIntegerConst :: SharedContext -> Integer -> IO Term
 scIntegerConst sc i
   | i >= 0    = scNatToInt sc =<< scNat sc (fromInteger i)
   | otherwise = scIntNeg sc =<< scNatToInt sc =<< scNat sc (fromInteger (- i))
 
--- primitive intAdd/intSub/intMul/intDiv/intMod :: Integer -> Integer -> Integer;
+-- | Create terms applying the (binary) integer primitives.
+--
+-- > intAdd/intSub/intMul/intDiv/intMod : Integer -> Integer -> Integer;
```

Comment:
Does this actually generate reasonable html documentation? It might be preferable to split this into separate type signatures for each function.